src/ucp/api/ucp.h

```diff
@@ -238,13 +238,15 @@ enum ucp_worker_attr_field {
  * The enumeration list describes the datatypes supported by UCP.
  */
 enum ucp_dt_type {
-    UCP_DATATYPE_CONTIG  = 0,      /**< Contiguous datatype */
-    UCP_DATATYPE_STRIDED = 1,      /**< Strided datatype */
-    UCP_DATATYPE_IOV     = 2,      /**< Scatter-gather list with multiple pointers */
-    UCP_DATATYPE_GENERIC = 7,      /**< Generic datatype with
-                                        user-defined pack/unpack routines */
-    UCP_DATATYPE_SHIFT   = 3,      /**< Number of bits defining
-                                        the datatype classification */
+    UCP_DATATYPE_CONTIG    = 0,      /**< Contiguous datatype */
+    UCP_DATATYPE_STRIDED   = 1,      /**< Interleaving pointers at fixed intervals */
+    UCP_DATATYPE_IOV       = 2,      /**< Scatter-gather list with multiple pointers */
+    UCP_DATATYPE_STRIDED_R = 3,      /**< Same as Strided, only reusable */
+    UCP_DATATYPE_IOV_R     = 4,      /**< Same as IOV, only reusable */
+    UCP_DATATYPE_GENERIC   = 7,      /**< Generic datatype with
```

Comment:
see "reusable" discussion above... i don't mind changing the name. what would you call a datatype handle to create once and pass to every send/recv call (for optimization, storing mkey and saving lookups), instead of re-creating before each call?